### PR TITLE
chore: fix warning - turn array default into function for `outsideElementClasses` prop

### PR DESCRIPTION
**Pull request by sareg0**
This fixes some warnings in the dev tools that impede local development. Currently we have many warnings when developing locally and it is hard to see what is really important and what is just a warning from a component. 

I deliberately left in the formatting changes, and have pointed out the exact changes for ease of reviewing. 

project.state.warnings includes many warning. In this PR I am fixing only the first one related to `outsideElementClasses`
```
[
  "TOPCOAT WARNING: Invalid default value for prop \"outsideElementClasses\": Props with type Object/Array must use a factory function to return the default value.",
  "TOPCOAT WARNING: The data property \"width\" is already declared as a prop. Use prop default value instead.",
  "TOPCOAT WARNING: Invalid prop: type check failed for prop \"numberValue\". Expected Number with value 0, got String with value \"\".",
  "TOPCOAT WARNING: Invalid prop: type check failed for prop \"numberPrevious\". Expected Number with value 0, got String with value \"\".",
  "TOPCOAT WARNING in library component 'TCsvExport': Unknown custom element: <basebutton> - did you register the component correctly? For recursive components, make sure to provide the \"name\" option.",
  "TOPCOAT WARNING in library component 'TDropdown': Invalid default value for prop \"outsideElementClasses\": Props with type Object/Array must use a factory function to return the default value.",
  "TOPCOAT WARNING in library component 'TFilterGroup': Invalid default value for prop \"outsideElementClasses\": Props with type Object/Array must use a factory function to return the default value.",
  "TOPCOAT WARNING in visualization 'TBigNumber': Invalid prop: type check failed for prop \"numberValue\". Expected Number with value 0, got String with value \"\".",
  "TOPCOAT WARNING in visualization 'TBigNumber': Invalid prop: type check failed for prop \"numberPrevious\". Expected Number with value 0, got String with value \"\".",
  "TOPCOAT WARNING: Invalid prop: type check failed for prop \"size\". Expected Number with value 18, got String with value \"18\"."
]
```
The output in the console is very verbose. See attached file for an example of the console output today, before this fix.
[console-export-2022-9-13_15-13-18.txt](https://github.com/topcoat-data/topcoat-public/files/9557396/console-export-2022-9-13_15-13-18.txt)

Below are screenshots of the dev tools before and after making the change (search term `outsideElementClasses`). Feel free to leave the search term off if you want.

### Before
Screenshot 2022-09-13 at 14 15 36

### After 
Screenshot 2022-09-13 at 14 13 07

### To test
Open the Issues Detail report locally, you should see warnings related to the `outsideElementClasses`

Change `config.yml` to point at this branch for topcoat-public
```
modules:
    - git: https://github.com/topcoat-data/topcoat-public.git
      revision: chore/fix-outsideElementClasses-warnings
```

Rebuild the modules 
Open the Issues Detail report locally, you should not see any warnings related to the `outsideElementClasses`

Check that the Date Picker in particular still works well, and closes when clicking outside of the element.

### CODE REVIEW

File: library/components/TDropdown.vue [hunk 1]
```diff
@@ -1,124 +1,126 @@
...
-                default: '999',
-            },
-			isExpanded: {
-				type: Boolean,
-				default: false,
-			},
-			outsideElementClasses: {
-				type: Array,
-				// Todo: Remoave & Replace with a better fix.
-				default: ["mx-datepicker-popup"],
```
Comment (sareg0) [inline library/components/TDropdown.vue:43]: before change

File: library/components/TDropdown.vue [hunk 2]
```diff
@@ -1,124 +1,126 @@
...
+      default: "999",
+    },
+    isExpanded: {
+      type: Boolean,
+      default: false,
+    },
+    outsideElementClasses: {
+      type: Array,
+      // Todo: Remove & Replace with a better fix.
+      default: () => ["mx-datepicker-popup"],
```
Comment (sareg0) [inline library/components/TDropdown.vue:43]: after change

File: library/components/TFilterGroup.vue [hunk 3]
```diff
@@ -1,227 +1,256 @@
...
-            type: String,
-            default: '400px'
-        },
-        isPersisted: {
-            type: Boolean,
-            default: false,
-        },
-        outsideElementClasses: {
-            type: Array,
-            default: [],
```
Comment (sareg0) [inline library/components/TFilterGroup.vue:108]: before change

File: library/components/TFilterGroup.vue [hunk 4]
```diff
@@ -1,227 +1,256 @@
...
-                            }
-                        } else {
-                            this.showInactive(componentInstance.tag_unique);
-                        }
-                    }
-                }
-            }
+    outsideElementClasses: {
+      type: Array,
+      default: () => [],
```
Comment (sareg0) [inline library/components/TFilterGroup.vue:133]: after change

**Review by gau94rav — APPROVED** [review]

**Pull request merged**